### PR TITLE
Some dictionary failures did not honor the user-provided reason

### LINKED ISSUE #1449: BeEquivalentTo does not report the because-phrase for dictionary mismatch
### Description

When dictionaries do not match, `BeEquivalentTo` does fail the test as expected, but it does **not** report the because-phrase text.

See fiddle https://dotnetfiddle.net/XUtEn5

### Complete minimal example reproducing the issue

```csharp
var actual = new Dictionary<string, string> { { "b", "x" } };
var expected = new Dictionary<string, string> { { "c", "x" } };

actual.Should().BeEquivalentTo(expected, "b-phrase");
```

### Expected behavior:

Logged output should include: _b-phrase_

### Actual behavior:

Logged output: _Expected actual to contain key "c"._

### Versions

* Which version of Fluent Assertions are you using? Tried with latest: 6.0.0.0-alpha0001, as well as 5.10.3. Same results in both.
* Which .NET runtime and version are you targeting? .NET Core 3.1.

### Additional Information

I discovered this issue at the same time: https://github.com/fluentassertions/fluentassertions/issues/1450

### PR DESCRIPTION
**Pull request by dennisdoomen**
Fixes #1449

### CODE REVIEW

File: docs/_pages/releases.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ sidebar:
 
 **Fixes**
 * Sometimes `BeEquivalentTo` reported an incorrect message when a dictionary was missing a key - [1454](https://github.com/fluentassertions/fluentassertions/pull/1454)
+* Some dictionary failures did not honor the user-provided reason - [1456](https://github.com/fluentassertions/fluentassertions/pull/1456)
```
Comment (jnyrup) [inline docs/_pages/releases.md:16]:
```suggestion
* Some dictionary failures did not honor the user-provided reason - [#1456](https://github.com/fluentassertions/fluentassertions/pull/1456)
```

**Review by jnyrup — APPROVED** [review]
Noticing that you've added `{reason}` to 6 places, but only updated 5 tests.

File: docs/_pages/releases.md (hunk 1 above)
Comment (dennisdoomen) [reply docs/_pages/releases.md:16]: I know. I couldn't find a test covering that test.

Comment (jnyrup) [reply docs/_pages/releases.md]: Fine with me, just wanted to check.

**Pull request merged**